src/main/java/frc/robot/component/Climber.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.component;
+
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkBase.SoftLimitDirection;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.units.Distance;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import frc.robot.encoder.MoEncoder;
+import frc.robot.util.MoPrefs;
+import frc.robot.util.MoShuffleboard;
+
+/* A single side climber */
+public class Climber {
+    public final CANSparkMax winch;
+    public final MoEncoder<Distance> encoder;
+
+    public final GenericEntry hasZero;
+
+    public Climber(String name, CANSparkMax winch) {
+        this.winch = winch;
+        winch.restoreFactoryDefaults();
+
+        this.encoder = MoEncoder.forSparkRelative(winch.getEncoder(), Units.Centimeters);
+
+        MoPrefs.climberEncoderScale.subscribe(this.encoder::setConversionFactor, true);
+        MoPrefs.climberZeroCurrentCutoff.subscribe(
+                current -> winch.setSmartCurrentLimit((int) current.in(Units.Amps)), true);
```

Comment:
The zeroing current threshold and the smartCurrentLimit should be entirely different values. The zeroing current should be a low threshold that we just use to zero when we're up against the hard stops. But the smartCurrentLimit is a very high threshold that should only trip if we're at risk of damaging the motor.

Think, if we're lifting the whole robot, we'll be using a lot of power and current. So the smartCurrentLimit should be high so the motor can actually provide the requisite power. But we don't want to apply all that power against the hard limits on the climber--we could break something! The zeroing current threshold should be a low value, still greater than the current used by climber when not loaded, but not as much as we'd use when actually climbing.

The zeroing current is a value we'll have to tune, so it should be a robot preference. But the smartCurrentLimit is a value inherent to the motor we're using (40-60A for a Neo, [per rev](https://docs.revrobotics.com/sparkmax/gs-sm/connect-a-spark-max-over-usb#limiting-current)), so it can stay a class constant.

Refer to the ArmSubsystem and IntakeSubsystem for examples.